File: app/models/address.rb
class Address < ApplicationRecord
  belongs_to :user

  validates :address_line1, :city, :state, :zipcode, :country, presence: true

  # Callbacks
  after_save :ensure_only_one_default, if: :is_default?

  # Scopes
  scope :default, -> { where(is_default: true) }

  def full_name
    [
      address_line1,
      address_line2,
      "#{city}, #{state} #{zipcode}",
      country
    ].compact.join("\n")
  end

  private

  # Ensure that only one address per user is marked as default
  def ensure_only_one_default
    if is_default?
      user.addresses.where.not(id: id).update_all(is_default: false)
    end
  end
end

File: frontend/src/types/address.ts
export interface Address {
  id: string;
  userId: string; // matches Rails user_id
  fullName: string; // matches Rails full_name
  addressLine1: string; // matches Rails address_line1
  addressLine2?: string; // matches Rails address_line2
  city: string;
  state: string;
  postalCode: string; // matches Rails postal_code
  country: string;
  phoneNumber: string; // matches Rails phone_number
  isDefault: boolean; // matches Rails is_default
}
